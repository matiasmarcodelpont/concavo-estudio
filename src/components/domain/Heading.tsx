import { cn } from '@/lib/utils'

interface HeadingProps {
  children: string
}

const Heading = ({ children, ...headingProps }: HeadingProps & React.HTMLAttributes<HTMLHeadingElement>) => {
  return (
    <h1
      {...headingProps}
      className={cn('uppercase text-4xl font-bold tracking-widest mt-12 mb-6', headingProps.className)}
    >
      {children}
    </h1>
  )
}

export default Heading
