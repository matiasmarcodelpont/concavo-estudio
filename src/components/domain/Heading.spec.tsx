import { render } from '@testing-library/react'

import Heading from './Heading'

describe('Heading', () => {
  it('matches snapshot', () => {
    const { getByRole } = render(<Heading>TÃ­tulo</Heading>)
    const component = getByRole('heading')

    expect(component).toMatchSnapshot()
  })
})
