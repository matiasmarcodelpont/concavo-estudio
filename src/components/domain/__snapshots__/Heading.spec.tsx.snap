// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading matches snapshot 1`] = `
<h1
  class="uppercase text-3xl mt-12 mb-6 sm:mt-16 sm:mb-9 md:mt-20 md:mb-12 italic opacity-0"
>
  TÃ­tulo
</h1>
`;
